addons/product/models/product_pricelist.py

```diff
@@ -162,6 +162,7 @@ def _compute_price_rule(self, products_qty_partner, date=False, uom_id=False):
         items = self.env['product.pricelist.item'].browse(item_ids)
         results = {}
         for product, qty, partner in products_qty_partner:
+            product = product.with_context(no_variant_attributes_price_extra=self._context.get('no_variant_attributes_price_extra'))
```

Comment:
To be checked why we don't need this in 13.0

Reply:
Just a guess here, need to dig deeper but maybe this has something to do with this new api of v13, perhaps in v12 since there is no enforced dependency the value is computed correctly and then overwritten when the context key is not set 
https://github.com/odoo/odoo/blob/a3d91cc8691794d8f93ef416830b09fabcf08b66/addons/product/models/product.py#L221-L222

Reply:
Yes, I just saw that too and it could be the cause that it works in v13.
In addition, I found out that there was a refactor of something linked: 22a11a6f4d0b08a56712faf

However, I put this line at the top of the loop but I think it would work (and be more secure) to put the context only there https://github.com/odoo-dev/odoo/blob/9bfca41925df92a692e936adc32e03210a2a37f5/addons/product/models/product_pricelist.py#L185

and there https://github.com/odoo-dev/odoo/blob/9bfca41925df92a692e936adc32e03210a2a37f5/addons/product/models/product_pricelist.py#L218

Reply:
indeed, it is probably better to keep the context modifications restricted to just where they are needed